hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -476,14 +502,34 @@ public FSDataOutputStream createNonRecursive(final Path f, final FsPermission pe
     TracingContext tracingContext = new TracingContext(clientCorrelationId,
         fileSystemId, FSOperationType.CREATE_NON_RECURSIVE, tracingHeaderFormat,
         listener);
+    /*
+     * Get exclusive access to folder if this is a directory designated for atomic
+     * rename. The primary use case of the HBase write-ahead log file management.
+     */
+    AbfsBlobLease abfsBlobLease = null;
```

Comment:
As currently v1 doesnt provide this, lets have this functionality in a config control and have it as off.

Reply:
Added for createNonRecursive lease acquire: fs.azure.lease.create.non.recursive. default: false